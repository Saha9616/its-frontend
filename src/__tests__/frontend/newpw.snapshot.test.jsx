import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ResetPasswordView from "@/app/(auth)/new-password/page";

// mock useSearchParams
jest.mock("next/navigation", () => ({
  useSearchParams: () => new URLSearchParams({ token: "mockToken" }),
}));

it("renders new password page unchanged", async () => {
  const result = await ResetPasswordView();
  const { container } = render(result);
  expect(container).toMatchSnapshot();
});
