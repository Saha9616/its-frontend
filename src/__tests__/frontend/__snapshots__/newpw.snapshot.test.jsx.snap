// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new password page unchanged 1`] = `
<div>
  <div
    class="flex flex-col space-y-8 justify-center items-center min-h-screen"
  >
    <h1
      class="text-2xl font-semibold"
    >
      ITS Frontend Manager
    </h1>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm min-w-96 md:min-w-[500px]"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight text-center"
        >
          New Password
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <form
          class="space-y-4"
        >
          <div
            class="space-y-2"
          >
            <label
              class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
              for=":r0:-form-item"
            >
              New Password
            </label>
            <div
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              class="flex space-x-2 items-center"
              id=":r0:-form-item"
            >
              <svg
                class="lucide lucide-lock "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="11"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="11"
                />
                <path
                  d="M7 11V7a5 5 0 0 1 10 0v4"
                />
              </svg>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="flex flex-col w-full justify-center items-center text-center"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-3/4 mt-4"
              type="submit"
            >
              Change Password
            </button>
          </div>
        </form>
        <div
          class="flex items-center p-6 pt-0 justify-center mt-4"
        >
          <span
            class="text-sm"
          >
            Found your password?
          </span>
          <a
            href="/signin"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2 text-sm"
            >
              Login now
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
